syntax = "proto3";

package kv_store;

service KeyValueStore {
  rpc Read(ReadRequest) returns (ReadResponse) {}
  rpc Write(WriteRequest) returns (WriteResponse) {}
}

message ReadRequest {
  string key = 1;
}

message ReadResponse {
  string value = 1;
}

message WriteRequest {
  string key = 1;
  string value = 2;
}

message WriteResponse {
  oneof response {
    string previous_value = 1;
    int32 master_id = 2;
  }
}
